src/components/structures/MatrixChat.js

```diff
@@ -518,6 +519,32 @@ module.exports = React.createClass({
             case 'set_theme':
                 this._onSetTheme(payload.value);
                 break;
+            case 'unknown_device_error':
```

Comment:
It could easily be a utility. It would be nicer to have a `DialogSpawner` component or something, but this wouldn't be visible. I'm slightly concerned about dumping this in `src`. I wished files were grouped by function (like the `autocomplete` and `login` directories) and not by category like `elements`.